docs/adr/0016-publishing-times-and-change-history.md

```diff
@@ -0,0 +1,208 @@
+input. This item has a change note of "First published." and has the time the
+content was initially published on GOV.UK.
+
+To set these metadata fields Content Publisher has relied on the Publishing API
+automatically setting the time fields and using it's
+[ChangeNote][pub-api-change-note] model. This model stores a record for each
+major change published on GOV.UK and can collate them to build a change
+history. This meant that Content Publisher has not needed to store
+`first_published_at`, `public_updated_at` or a full change history and has only
+needed to store a change note for an edition.
+
+We had concerns that continued usage of the Publishing API time and change
+history system would lead to discrepancies when migrating content from Whitehall
+to Content Publisher. This led to us doing a review of the system and problems,
+where some relate to migration and others are broader.
+
+### Problems identified
+
+#### 1. Having accurate times for Whitehall migrated content
+
+When content is migrated from Whitehall to Content Publisher we want all the
+times associated with that content to remain the same. Whitehall complicates
+this by passing two time values to represent the first publishing time: the
+aforementioned `first_published_at` field and a deprecated time value,
+`first_public_at`, which is used [preferentially to
+`first_published_at`][first-public-at] when rendering the content. Content
+Publisher does not populate the `first_public_at` field.
+
+An occasional problem is that the Publishing API can have different values for
+each of these times. A consequence of these not matching is that migrated
+content would show a different first publishing time.
+
+#### 2. Migrating change notes from Whitehall
+
+Whitehall doesn't make use of the [Publishing API ChangeNote][pub-api-change-note]
+model to build a change history, instead Whitehall creates it's own change
+history for content and includes it when updating content with the Publishing
+API. This has a consequence that the data stored in ChangeNote models in
+Publishing API for Whitehall content cannot be relied upon as there are no
+guarantees of its accuracy. There is also not an expectation that this data is
+accurate as many pieces of Whitehall content were created before the
+ChangeNote model was introduced to Publishing API.
+
+For Content Publisher to migrate change histories for Whitehall content it
+would need to ensure the entire change history is in sync with the Publishing
+API.
+
+#### 3. Removing/editing change notes
+
+It is a relatively common task that GOV.UK 2nd line support engineers are
+[requested to remove or modify a change note][change-note-docs] for a published
+piece of content. To resolve this in Whitehall a developer would modify a past
+edition of a piece of content to either remove or amend the change note.
+
+In Content Publisher it is not ideal to modify a past edition. An edition
```

Comment:
Minor: we could link to the modelling ADR to provide clear support for this view.

Reply:
Added in.